assets/app/view/game/buy_trains.rb

```diff
@@ -302,7 +302,7 @@ def from_depot(depot_trains, corporation)
             president_assist, _fee = @game.president_assisted_buy(@corporation, train, price)
             entity = @corporation
 
-            if @selected_company&.owner == @corporation
+            if @selected_company&.owner == @corporation || @selected_company&.owner == @corporation.owner
```

Comment:
Is [@corporation, @corporation.owner].include?(@selected_company&.owner) more DRY? Up to you.